packages/babel-core/src/transform-file.js

```diff
@@ -42,6 +42,20 @@ const transformFileRunner = gensync<[string, ?InputOptions], FileResult | null>(
     return yield* run(config, code);
   },
 );
+export const transformFile: Transform = (function transform(code, opts, callback) {
+  if (typeof opts === "function") {
+    callback = opts;
+    opts = undefined;
+  }
+
+  if (callback === undefined) {
+    throw new Error(
+      "Starting from Babel 8.0.0, the 'transformFile' function expects a callback. If you need to call it synchronously, please use 'transformFileSync",
+    );
+  }
+
+  transformFileRunner.errback(code, opts, callback);
+}: Function);
 
 export const transformFile: TransformFile = transformFileRunner.errback;
```

Comment:
Can you delete this line? Otherwise `transformFile` is declared twice.

Reply:
Yes. Thanks.